druid/src/window.rs

```diff
@@ -426,6 +426,11 @@ impl<T: Data> Window<T> {
         for &r in invalid.rects() {
             piet.clear(
                 Some(r),
+                // this if statement, and the transparent setting in general
+                // should be unnecessary, simply setting the alpha channel on the
+                // window background color < 1.0 should be sufficient to enable
+                // transparency. this logic remains until all backends
+                // support such a behavior.
                 if self.transparent {
```

Comment:
I think this is not the correct way to implement this. I would just: 
```suggestion
                if self.transparent && env.get(crate::theme::WINDOW_BACKGROUND_COLOR).as_rgba8().3 == 255
```

Reply:
that won't work. I'm looking or semi transparency for any color which given the current implementation requires the backends to change.

Reply:
or example:
![screenshot-2021-09-16-104353](https://user-images.githubusercontent.com/2835871/133633216-652ca201-56af-43fe-84ef-d5fab17f258c.png)

Reply:
that should work without backend change. just set background color with some alpha and `.transparent(true)`.

Reply:
I did complement changing that code to use the backendground color. but then the logic gets really convoluted.
as now you need to not only check for the transparency flag, but then you need to get the background color, check if its alpha is set, otherwise default its alpha to 0.

as a result I had the choice of:
1. writing more conditional logic
2. just removing code.

option 2 is clearly the better option.

Reply:
possibly, but I'd wait for that to actually a real issue before complicating the API on top of everything. blending layers is a pretty common path through compositors.

and if the performance issue ever did come up in a real use case, I'd push for the logic to still be pushed down to the backends, as this is still simpler both in terms of end user API, code complexity.

```
        let mut transparent = self.background.3 < 0xFF;
        if let Some(screen) = window.screen() {
            let visual = screen.rgba_visual();
            transparent = visual.is_some();
            window.set_visual(visual.as_ref());
        }
```

Reply:
oh and just to be clear, I'll happily change this to whatever you want in the end. just pointing out why I didn't just make the change you suggested originally. mainly due to the minor irritation caused by the surprise of needing to set an additional flag on top of the color itself.

just let me know which approach you want me to take. ;)

Reply:
your approach sound reasonable. We should deprecate `set_transparent`(because it is useless now)  . if we see performance issues in futures, we can move background color to the backends

Reply:
also: we can just `transparent = true` default. then user can set transparent false if they are having performance problems

Reply:
and changing the default should be a simpler change

Reply:
@Maan2003 updated.